src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -147,10 +152,18 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol e
                 conditionPartToCheck.GetLocation(),
                 whenPartToCheck.GetLocation());
 
+            var properties = ImmutableDictionary<string, string>.Empty;
+            var whenPartIsNullable = semanticModel.GetTypeInfo(whenPartMatch).Type?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
```

Comment:
Sounds like we should not run analyzers prior to references getting resolved.  Or we should invest in a large audit pass for that occurrence happening.  THe design for analyzers was that they would run on an accurate snapshot of the system.  meaning if you had defined references and source files that all data would be accurate across those definitions.  It now sounds like analyzers can observe a state where references aren't actually properly connected, and they've never been designed to handle that.